web/modules/custom/rdf_entity_provenance/rdf_entity_provenance.module

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * @file
+ * Contains hooks and functions for the rdf_entity_provenance module.
+ */
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+
+/**
+ * Implements hook_entity_bundle_field_info().
+ */
+function rdf_entity_provenance_entity_bundle_field_info(EntityTypeInterface $entity_type, $bundle, array $base_field_definitions) {
+  if ($entity_type->id() !== 'rdf_entity' || $bundle !== 'provenance_activity') {
+    return [];
+  }
+
+  $fields = [];
+
+  // The time that the activity started to federate the remote entity. This is
+  // different from the created date as it can be updated if the entity is
+  // federated again.
+  // @see: https://www.w3.org/TR/2013/REC-prov-o-20130430/#startedAtTime
+  $fields['provenance_started'] = BaseFieldDefinition::create('timestamp')
```

Comment:
I wonder why we don't map the `changed` base field to `http://www.w3.org/ns/prov#startedAtTime` and get rid of this field. The advantage is that, we get the default value automatically. For example, with the actual code, before saving an entity created with `ProvenanceHelper::createProvenanceActivity()` we need to assign explicitly the `provenance_started`. But `changed` does it automatically, so when saving we'll get the correct time without worrying.

Reply:
Because normally, the provenance entity has both `provenance_started` and `provenance_ended`. The timestamp here should take the time that the import started, not the one that the entity was saved. The time difference is of course not of vital value nor will diverge a lot, but for example, if we were to import thousands of entities, we would fetch them, we would parse them and we would store them. In this case, a time difference exists between the time that the fetching started (provenance_started) until the conversion/validation and finally the saving of the entity.